### PR TITLE
[DSOUND_NEW] Add CMakeLists.txt + fix gcc compilation

### PR DESCRIPTION
**Pull request by oleg-dubinskiy**
## Purpose
Add CMakeLists.txt and mostly fix gcc compilation of dsound_new. Although it is still not compilable, my PR fixes at least part of compilation issues.
If you want to test it, you need to comment out `add_subdirectory(dsound)` in dll/directx/wine/CMakeLists.txt and uncomment `#add_subdirectory(dsound_new) # disabled in trunk` in dll/directx/CMakeLists.txt.

JIRA issue: [CORE-16128](https://jira.reactos.org/browse/CORE-16128)

### CODE REVIEW

File: dll/directx/dsound_new/precomp.h [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#pragma once
+#ifndef PRECOMP_H
+#define PRECOMP_H
```
Comment (learn-more) [inline dll/directx/dsound_new/precomp.h]: Why is this change needed?
gcc should support `#pragma once` ?

Comment (oleg-dubinskiy) [reply dll/directx/dsound_new/precomp.h]: Yes, same as other compilers. But for some reason, at least RosBE Unix 2.1.2 can't compile it with `#pragma once`. I get an error "#pragma once in main file" and compilation doesn't go any further. :(
As far as I know, `#pragma once` works with more number of compilers than "#ifndef/#define/#endif", but is more farther from C/C++ standards.

Comment (ThFabba) [reply dll/directx/dsound_new/precomp.h]: `#pragma once` doesn't work in precompiled header files.

File: dll/directx/dsound_new/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    precomp.h
+    resource.h)
+
+add_library(dsound MODULE
+    ${SOURCE}
+    version.rc)
+
+set_module_type(dsound win32dll)
+target_link_libraries(dsound dxguid uuid)
+add_importlibs(dsound winmm advapi32 msvcrt kernel32 ntdll)
```
Comment (feel-the-dz3n) [inline dll/directx/dsound_new/CMakeLists.txt]: add ``ole32`` here, I suppose

Comment (oleg-dubinskiy) [reply dll/directx/dsound_new/CMakeLists.txt]: Indeed! After adding `ole32` it does no longer spam about `CoTaskMemFree@4`! By the same principle I added other needed importlibs into `add_importlibs()` section: `ksuser`, `setupapi` and `user32`. And now compilation succeeds! Thanks for spotting! :smile:
